source/agora/node/Registry.d

```diff
@@ -33,13 +33,17 @@ import agora.stats.Registry;
 import agora.stats.Utils;
 import agora.utils.Log;
 
+import vibe.core.core;
+import vibe.core.net;
```

Comment:
That's the big issue. The reason those were separated was to allow for dependency injection.
Having a straight `{TCP,UDP}Connection` in the registry means that we loose the nice separation we had between the business code and the IO code.

Reply:
I get it. We will need DNS client to ask queries to an another server within a DNS server. For example, AXFR is one of them, which also requires TCP. I thought that whole name registry in a single packet would be easier to manage.
Apart from TCP/UDP, we have REST (HTTP) interface implemented within registry itself.

Reply:
Yes, we need the client to be abstracted as well, but I don't understand why the refactoring here, as it's server side ?

Reply:
I was going to implement client into Registry also 😃 Closing PR now